src/qibo/transpiler/optimizer.py

```diff
@@ -27,7 +27,10 @@ def __call__(self, circuit: Circuit) -> Circuit:
             )
         if logical_qubits == physical_qubits:
             return circuit
-        new_circuit = Circuit(physical_qubits)
+        new_wire_names = circuit.wire_names.copy() + [
+            name for name in self.connectivity.nodes if name not in circuit.wire_names
+        ]
```

Comment:
```suggestion
        new_wire_names = circuit.wire_names + list(self.connectivity.nodes - circuit.wire_names)
```

`nx.NodeView` object seems to support set operations (union, intersection, ... and in particular sets' difference).
Moreover, `list.copy()` is a shallow copy. Such a shallow copy is also provided by plain addition.
```py
[ins] In [1]: x = [0, 1, 2]
[ins] In [2]: y = x + [4]
[ins] In [3]: del y[0]
[ins] In [4]: y
Out[4]: [1, 2, 4]                                                                                                                                                                           
[ins] In [5]: x
Out[5]: [0, 1, 2]
```
(of course, it's shallow, so deep modifications within the list elements are shared, but that's the same with `.copy()`)

Reply:
[f19dc6e](https://github.com/qiboteam/qibo/pull/1500/commits/f19dc6ed1ecb1e624ba6c4bbc63f72c3d3f56f20)